lib/basedriver/commands/execute.js

```diff
@@ -0,0 +1,102 @@
+import _ from 'lodash';
+import log from '../logger';
+import vm from 'vm';
+import { attach } from 'webdriverio';
+import { MJSONWP_ELEMENT_KEY, W3C_ELEMENT_KEY } from '../../protocol/protocol';
+
+const BASE_PATH = '/wd/hub'; // TODO defining this here is brittle but it is hardcoded in routes, so...
+
+let commands = {};
+
+/*
+ * TODO fill out entire docstring
+ *
+ * This method takes a string which is executed as javascript in the context of
+ * a new nodejs VM, and which has available a webdriverio driver object, having
+ * already been attached to the currently running session.
+ */
+commands.executeDriverScript = async function (script, scriptType) {
+  if (!!scriptType && scriptType !== 'webdriverio') {
+    throw new Error("Only the 'webdriverio' script type is currently supported");
+    // TODO should probably add support for WD script type as well?
+  }
+
+  if (!this.opts.host || !this.opts.port) {
```

Comment:
isn't the host always equal to localhost?

Reply:
it doesn't have to be, you can set it with the `-h` parameter.

Reply:
Hmm, I'm not sure what's going on here. If you increase the timeout does it just wait until the timeout? 1s is a pretty low timeout. The issue appears to be that the XCUITest driver is proxying the /timeouts call to WDA, but WDA can't handle it. Probably /timeouts needs to be added to NO_PROXY in WDA?

But the more concerning issue is why we got a timeout error and not the "Unhandled Endpoint" error which we would have expected. That's why I'm asking you to increase the timeout in case it's just a race condition.

Reply:
> Will you work to set this.opts.host and this.opts.port in appium repo?

I don't understand. My assumption was that `this.opts.host` and `this.opts.port` are always set because the main Appium server copies its args on to `InnerDriver.opts`, is that not true?

Reply:
> this.opts.host and this.opts.port are always set

mm, let me take a look further later.
On my environment, 

1. `run clean` in appium repo
2. Link the node_module/appium-base-driver to this PR
3. Launch the appium
4. Run client script as xcuitest

Then, _Host or port of running server were not defined_ error was raised. I set `this.opts.host = 'localhost'; this.opts.port = '4723';` before this line, it could avoid. So, I thought this option was new items.
I did not check other driver cases.

Reply:
were you starting an Appium server or just a XCUITest driver server? The latter will not work; only the main Appium server copies the CLI args to the inner drivers.

Reply:
I ran appium server with `node . --relaxed-security` in appium repo (and xcuitest was automationName)
ah, did I miss some server arguments?

Reply:
ah, what about `opts.address` instead of `opts.host`?
Let me check...
https://github.com/appium/appium/search?q=address&unscoped_q=address

Reply:
When I replaced `opts.host` with `opts.address` in _lib/basedriver/commands/execute.js_, it worked fine with Python client script.
I could configure them as `-p 4723 -a localhost` [server arg](https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/server-args.md).

Could you double check about _address_ or _host_?

Reply:
OK, i ran into this too. I guess it's a bad assumption.